channel_test.go

```diff
@@ -112,3 +113,18 @@ func TestIsolatedSubChannelsDontSharePeers(t *testing.T) {
 	assert.NotNil(t, sub.peers.peersByHostPort["127.0.0.1:3000"])
 	assert.Nil(t, isolatedSub.peers.peersByHostPort["127.0.0.1:3000"])
 }
+
+func TestChannelTracerMethod(t *testing.T) {
+	tracer := mocktracer.New()
+	ch, err := NewChannel("svc", &ChannelOptions{
+		Tracer: tracer,
+	})
+	require.NoError(t, err)
+	defer ch.Close()
+	assert.Equal(t, tracer, ch.Tracer(), "expecting tracer passed at initialization")
+
+	ch, err = NewChannel("svc", &ChannelOptions{})
+	require.NoError(t, err)
+	defer ch.Close()
+	assert.EqualValues(t, opentracing.GlobalTracer(), ch.Tracer(), "expecting default tracer")
```

Comment:
Definitely agree it's a little iffy, but we don't run our test in parallel (none of them use the `t.Parallel`). If we do decide to use it, we'll have to be careful not to enable it on this test (or other tests which rely on global state). I think it's better to have the test and validation for now, thanks for adding the test.